compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangParserListener.java

```diff
@@ -950,6 +951,48 @@ public void exitRecordKey(BallerinaParser.RecordKeyContext ctx) {
         }
     }
 
+    @Override
+    public void enterTableLiteral(BallerinaParser.TableLiteralContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+        this.pkgBuilder.startTableLiteral();
+    }
+
+    @Override
+    public void exitTableColumn(BallerinaParser.TableColumnContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+        String columnName = ctx.getChild(0).getText();
+        boolean keyColumn = KEYWORD_PRIMARYKEY.equals(columnName);
+        if (keyColumn) {
+            columnName = ctx.getChild(1).getText();
```

Comment:
Is no of children always greater than 0 here?

Reply:
Yes. Otherwise parse error is thrown before get in to this